Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -24,6 +30,132 @@ namespace AzToolsFramework
+            }
+
+            if (AZStd::optional<AZStd::string_view> localBookmarksId = AZ::StringFunc::TokenizeLast(path, "/");
+                localBookmarksId.has_value() && path == m_viewBookmarksKey && !localBookmarksId->empty())
+            {
+                auto setVec3Fn = [value](AZ::Vector3& inout, int currentIndex)
+                {
+                    switch (currentIndex)
+                    {
+                    case 0:
+                        inout.SetX(aznumeric_cast<float>(value));
+                        break;
+                    case 1:
+                        inout.SetY(aznumeric_cast<float>(value));
+                        break;
+                    case 2:
+                        inout.SetZ(aznumeric_cast<float>(value));
+                        break;
+                    default:
+                        AZ_Warning(
+                            "LocalViewBookmarkLoader", false, "Trying to set an invalid index in a Vector3, index = %d", currentIndex);
+                        break;
+                    }
+                };
+
+                if (bookmarkType == LastKnownLocationKey)
+                {
+                    if (!m_lastKnownLocation.has_value())
+                    {
+                        m_lastKnownLocation.emplace(ViewBookmark{});
+                    }
+
+                    const int currentIndex = AZStd::stoi(AZStd::string(valueIndex));
+                    if (dataType == "Position")
+                    {
+                        setVec3Fn(m_lastKnownLocation->m_position, currentIndex);
+                    }
+                    else if (dataType == "Rotation")
+                    {
+                        setVec3Fn(m_lastKnownLocation->m_rotation, currentIndex);
+                    }
+                }
+                else if (bookmarkType == LocalBookmarksKey)
+                {
+                    if (auto existingBookmarkEntry = m_bookmarkMap.find(localBookmarksId.value());
+                        existingBookmarkEntry != m_bookmarkMap.end())
+                    {
+                        AZStd::vector<ViewBookmark>& bookmarks = existingBookmarkEntry->second;
+                        // if it is the first bookmark and it is the Position data it means it is the first one
+                        // and we have to create the Bookmark.
+                        if (valueIndex == "0" && dataType == "Position")
+                        {
+                            ViewBookmark bookmark;
+                            bookmark.m_position.SetX(aznumeric_cast<float>(value));
+                            bookmarks.push_back(bookmark);
+                        }
+                        else
+                        {
+                            const int bookmarkIndex = stoi(AZStd::string(bookmarkIndexStr->data()));
+                            AZ_Assert(bookmarkIndex < bookmarks.size(), "Bookmark index is out of bounds");
```

Comment:
Since `bookmarkIndex` is an integer and comes from converting a string, it could be possible to return a negative number.

Better to check that in the assert too `AZ_Assert(bookmarkIndex  >= 0 && bookmarkIndex < bookmarks.size(), ...)`

Reply:
Yup sure I can add this (again I didn't actually touch this code much just moved it from another function)